packages/ui-react/src/components/defi/protocol-tvl-historical-chart.tsx

```diff
@@ -0,0 +1,175 @@
+import { useQuery } from "@tanstack/react-query";
+import { useMemo } from "react";
+import { CartesianGrid, Line, LineChart, XAxis, YAxis } from "recharts";
+import {
+	type ChartConfig,
+	ChartContainer,
+	ChartTooltip,
+	ChartTooltipContent,
+} from "#components/shadcn/chart";
+import { Skeleton } from "#components/shadcn/skeleton";
+
+type ProtocolTvlHistoricalChartProps = {
+	protocol: string;
+	title?: string;
+	color?: string;
+};
+
+interface ProtocolTvlDataPoint {
+	date: number;
+	totalLiquidityUSD: number;
+}
+
+interface ChainTvlData {
+	[chainId: string]: ProtocolTvlDataPoint[];
+}
+
+interface ProtocolData {
+	chainTvls: ChainTvlData;
+	tvl: ProtocolTvlDataPoint[];
+	[key: string]: unknown;
+}
+
+const chartConfig = {} satisfies ChartConfig;
+
+async function getProtocolTvlHistoricalChartData(
+	protocolId: string,
+): Promise<ProtocolTvlDataPoint[]> {
+	const response = await fetch(`https://api.llama.fi/protocol/${protocolId}`);
+
+	if (!response.ok) {
+		throw new Error(
+			`Failed to fetch protocol TVL data: ${response.statusText}`,
+		);
+	}
+
+	const data = (await response.json()) as ProtocolData;
+
+	console.log(data);
+
+	// If chainName is specified, only include data for that chain
+	if (data.tvl) {
+		const chainData = data.tvl;
+		const validPoints = chainData.filter(
+			(point) =>
+				typeof point === "object" &&
+				"date" in point &&
+				"totalLiquidityUSD" in point,
+		);
+		return validPoints.sort((a, b) => a.date - b.date);
+	}
+
+	// Otherwise combine all valid chain data into a single dataset
+	const tvlData: ProtocolTvlDataPoint[] = [];
+
+	Object.values(data.chainTvls).forEach((chainData) => {
+		// Only include data where the required format is present
+		if (Array.isArray(chainData) && chainData.length > 0) {
+			const validPoints = chainData.filter(
+				(point) =>
+					typeof point === "object" &&
+					"date" in point &&
+					"totalLiquidityUSD" in point,
+			);
+			tvlData.push(...validPoints);
+		}
+	});
+
+	// Sort by date
```

Comment:
refer naming the code (e.g. .sort(sortWithDate) over explicit comment if necessary